pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -550,37 +550,52 @@ func TestReconcile(t *testing.T) {
 		err = r.client.Update(context.TODO(), userAcc)
 		require.NoError(t, err)
 
+		// Mock finalizer removal failure
+		fakeClient.MockUpdate = func(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error {
+			finalizers := []string{userAccFinalizerName}
+			userAcc := obj.(*toolchainv1alpha1.UserAccount)
+			userAcc.Finalizers = finalizers
+			return fmt.Errorf("unable to remove finalizer for user account %s", userAcc.Name)
+		}
+
+		res, err = r.Reconcile(req)
+		require.NoError(t, err)
+		assert.Equal(t, reconcile.Result{}, res)
+
+		res, err = r.Reconcile(req)
```

Comment:
OK if you still want to use multiple reconciles in the same test then at least check the status and relevant resources between them (in all tests).
For example the status should be "Terminating" and finalizer is not removed until the last reconcile.

Reply:
fixed